scripts_future_API/tm_stokes_mpi_wip.jl

```diff
@@ -13,53 +13,60 @@ const TBC = BoundaryCondition{Traction}
 const SBC = BoundaryCondition{Slip}
 
 using KernelAbstractions
+using AMDGPU
 
-using GLMakie
+using CairoMakie
 
 using FastIce.Distributed
 using MPI
 
+println("import done")
+
+@views avx(A) = 0.5 .* (A[1:end-1, :, :] .+ A[2:end, :, :])
+@views avy(A) = 0.5 .* (A[:, 1:end-1, :] .+ A[:, 2:end, :])
+@views avz(A) = 0.5 .* (A[:, :, 1:end-1] .+ A[:, :, 2:end])
+
 MPI.Init()
 
-backend = CPU()
-dims    = (2, 0, 2)
+backend = ROCBackend()
+dims    = (0, 0, 0)
 arch    = Architecture(backend, dims, MPI.COMM_WORLD)
 
 # physics
 ebg = 1.0
 
 topo = details(arch)
 
-size_l = (64, 64, 64)
+size_l = (128, 128, 128)
```

Comment:
For good perf shouldn't we have `(127, 127, 127)` here?

Reply:
Rather `(126, 126, 126)`, since we have `(nx+2, ny+2, nz+2)` stress kernels

Reply:
Oh ok - yeah.